src/node/internal/internal_http_incoming.ts

```diff
@@ -60,86 +94,335 @@ export class IncomingMessage extends Readable implements _IncomingMessage {
     });
 
     this.on('timeout', () => {
-      this.#reading = false;
+      this._consuming = false;
     });
 
-    this.#reader = this.#response.body?.getReader();
-    // eslint-disable-next-line @typescript-eslint/no-floating-promises
-    this.#tryRead();
+    this._stream = this.#response.body;
   }
 
   async #tryRead(): Promise<void> {
-    if (!this.#reader || this.#reading || this.#aborted) return;
-
-    this.#reading = true;
+    if (this._stream == null) return;
 
+    this.#reader ??= this._stream.getReader();
     try {
-      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-      while (true) {
-        const { done, value } = await this.#reader.read();
-        if (done) {
-          break;
-        }
-        this.push(value);
+      const data = await this.#reader.read();
```

Comment:
I'd move the `getReader()` call into the `try` here. getReader() can throw and that should not be ignored and should lead to `this` being destroyed.

```suggestion
      this.#reader ??= this._stream.getReader();
      const data = await this.#reader.read();
```